### PR TITLE
UX: Add classes to create-account modal for easier customization

### PR DESCRIPTION
**Pull request by tshenry**
This will follow the same pattern that was used for the associate-link and email input-group elements.

It also adds a class to the create-account-form element that highlights which auth provider is used if any.

Screen Shot 2022-07-11 at 3 02 46 PM

If no auth provider is used, no class is added.

### CODE REVIEW

File: app/assets/javascripts/discourse/app/templates/modal/create-account.hbs [hunk 1]
```diff
@@ -28,7 +28,7 @@
                 <span class="more-info">{{i18n "user.email.instructions"}}</span>
               </div>
 
-              <div class="input-group">
+              <div class="input-group create-account-username">
```
Comment (Chapoi) [inline app/assets/javascripts/discourse/app/templates/modal/create-account.hbs:31]:
```suggestion
              <div class="input-group create-account__username">
```

File: app/assets/javascripts/discourse/app/templates/modal/create-account.hbs [hunk 2]
```diff
@@ -41,7 +41,7 @@
                 <span class="more-info">{{i18n "user.username.instructions"}}</span>
               </div>
 
-              <div class="input-group">
+              <div class="input-group create-account-fullname">
```
Comment (Chapoi) [inline app/assets/javascripts/discourse/app/templates/modal/create-account.hbs:44]:
```suggestion
              <div class="input-group create-account__fullname">
```

File: app/assets/javascripts/discourse/app/templates/modal/create-account.hbs [hunk 3]
```diff
@@ -66,7 +66,7 @@
                     authOptions=this.authOptions
                   }} />
 
-              <div class="input-group">
+              <div class="input-group create-account-password">
```
Comment (Chapoi) [inline app/assets/javascripts/discourse/app/templates/modal/create-account.hbs:69]:
```suggestion
              <div class="input-group create-account__password">
```

File: app/assets/javascripts/discourse/app/templates/modal/create-account.hbs [hunk 4]
```diff
@@ -91,7 +91,7 @@
               </div>
 
               {{#if this.requireInviteCode }}
-                <div class="input-group">
+                <div class="input-group create-account-invite-code">
```
Comment (Chapoi) [inline app/assets/javascripts/discourse/app/templates/modal/create-account.hbs:94]:
```suggestion
                <div class="input-group create-account__invite-code">
```

**Review by Chapoi — CHANGES_REQUESTED** [review]
BEM-ified it :)

**Comment by tshenry** [comment]
@Chapoi I made the suggested changes, but also made the other `input-group` elements follow the same BEM pattern. Do you think that's okay at this point? Did you have a game plan for gracefully transitioning existing classes to follow BEM methodology? If it's better to not make changes to the existing classes, I can drop those changes.

**Comment by Chapoi** [comment]
> @Chapoi I made the suggested changes, but also made the other `input-group` elements follow the same BEM pattern. Do you think that's okay at this point? Did you have a game plan for gracefully transitioning existing classes to follow BEM methodology? If it's better to not make changes to the existing classes, I can drop those changes.

I think for existing classes we won't touch them for the foreseeable future, to avoid backwards compatibility issues. So think it's okay to keep those as-is, but good instinct though! :)

**Review by Chapoi — APPROVED** [review]

**Review by jordanvidrine — APPROVED** [review]

**Pull request merged**